analyzers/src/SonarAnalyzer.CSharp/Rules/UriShouldNotBeHardcoded.cs

```diff
@@ -21,27 +21,29 @@
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
-    public sealed class UriShouldNotBeHardcoded
-        : UriShouldNotBeHardcodedBase<ExpressionSyntax, LiteralExpressionSyntax,
-            SyntaxKind, BinaryExpressionSyntax, ArgumentSyntax, VariableDeclaratorSyntax>
+    public sealed class UriShouldNotBeHardcoded : UriShouldNotBeHardcodedBase<SyntaxKind, LiteralExpressionSyntax, ArgumentSyntax>
     {
-        private static readonly DiagnosticDescriptor Rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
-        protected override SyntaxKind StringLiteralSyntaxKind { get; } = SyntaxKind.StringLiteralExpression;
-        protected override SyntaxKind[] StringConcatenateExpressions { get; } = { SyntaxKind.AddExpression };
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
         protected override GeneratedCodeRecognizer GeneratedCodeRecognizer => CSharpGeneratedCodeRecognizer.Instance;
-        protected override string GetLiteralText(LiteralExpressionSyntax literalExpression) => literalExpression?.Token.ValueText;
-        protected override ExpressionSyntax GetLeftNode(BinaryExpressionSyntax binaryExpression) => binaryExpression.Left;
-        protected override ExpressionSyntax GetRightNode(BinaryExpressionSyntax binaryExpression) => binaryExpression.Right;
-        protected override bool IsInvocationOrObjectCreation(SyntaxNode node) =>
-            node.IsKind(SyntaxKind.InvocationExpression)
-            || node.IsKind(SyntaxKind.ObjectCreationExpression);
+        protected override SyntaxKind[] StringConcatenateExpressions => new[] { SyntaxKind.AddExpression };
+        protected override SyntaxKind[] InvocationOrObjectCreationKind => new[] { SyntaxKind.InvocationExpression, SyntaxKind.ObjectCreationExpression };
 
-        protected override int? GetArgumentIndex(ArgumentSyntax argument) =>
-            (argument?.Parent as ArgumentListSyntax)?.Arguments.IndexOf(argument);
+        protected override string GetLiteralText(LiteralExpressionSyntax literalExpression) =>
+            literalExpression?.Token.ValueText;
 
-        protected override string GetDeclaratorIdentifierName(VariableDeclaratorSyntax declarator) =>
-            declarator?.Identifier.ValueText;
+        protected override SyntaxNode GetRelevantAncestor(SyntaxNode node)
```

Comment:
I have already mentioned a proposal to simplify the implementation of this method in the VB.NET implementation.
Looking now at the C# version of it, I am wondering whether it can be of general use, and if so, if it should abstracted away from here, and moved to the base class or to the relevant facade. 
Once simplified, however, I think it may not be needed/useful anymore, since the method would become a one-liner...
To be checked.